lib/hardware_services/inventory_drivers/MockInventoryDriver.py

```diff
@@ -151,14 +151,47 @@ def load_data(self, quadsinstance, force):
             quadsinstance.logger.error(ex)
             exit(1)
 
-    def write_data(self, quadsinstance, doexit = True):
-        quadsinstance.quads_write_data_(doexit)
+    def write_data(self, quadsinstance, doexit):
+        try:
+            stream = open(quadsinstance.config, 'w')
```

Comment:
you are opening a file here but not closing it. better way to do it would be opening your file using `with`.
```python
with open(quadinstance.config) as stream:
       #your code
```
it's safer that way. and `with` takes care of closing the file for you so you don't need to explicitly close it.